src/test/test-models/good/integrate_ode_adams.stan

```diff
@@ -0,0 +1,49 @@
+functions {
+  real[] sho(real t,
+             real[] y,
+             real[] theta,
+             real[] x,
+             int[] x_int) {
+    real dydt[2];
+    dydt[1] = y[2];
+    dydt[2] = -y[1] - theta[1] * y[2];
+    return dydt;
+  }
+}
+data {
+  int<lower=1> T;
+  real y0_d[2];
+  real t0;
+  real ts[T];
+  real theta_d[1];
+  real x[0];
+  int x_int[0];
+}
+parameters {
+  real y0_p[2];
+  real theta_p[1];
+}
+model {
+  real y_hat[T,2];
+  y_hat = integrate_ode_adams(sho, y0_d, t0, ts, theta_d, x, x_int);
+  y_hat = integrate_ode_adams(sho, y0_d, t0, ts, theta_p, x, x_int);
+  y_hat = integrate_ode_adams(sho, y0_p, t0, ts, theta_d, x, x_int);
+  y_hat = integrate_ode_adams(sho, y0_p, t0, ts, theta_p, x, x_int);
+
+  y_hat = integrate_ode_adams(sho, y0_d, t0, ts, theta_d, x, x_int, 1e-10, 1e-10, 1e8);
+  y_hat = integrate_ode_adams(sho, y0_d, t0, ts, theta_p, x, x_int, 1e-10, 1e-10, 1e8);
+  y_hat = integrate_ode_adams(sho, y0_p, t0, ts, theta_d, x, x_int, 1e-10, 1e-10, 1e8);
+  y_hat = integrate_ode_adams(sho, y0_p, t0, ts, theta_p, x, x_int, 1e-10, 1e-10, 1e8);
+}
+generated quantities {
+  real y_hat[T,2];
+  y_hat = integrate_ode_adams(sho, y0_d, t0, ts, theta_d, x, x_int);
```

Comment:
These tests are for instantiations of functions in the language, not for assignment.  

The data argument versions should stay in generated quantities---that tests that the result is `double`.  They can be removed from transformed parameters as that just tests that a `double` can be promoted to `var` under assignment, which is another module's responsibility.

Reply:
Removed data argument tests in transformed parameters.